<a href="https://colab.research.google.com/github/SnigdhaPanakanti/World_Development_Explorer/blob/main/Part_B_Draft.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Individual Project PART-B**

**Name: Snigdha Panakanti**

## A Report comparing Food production, Agricultural land and Food insecurities in people.

**INTRODUCTION:**

Agriculture provides the largest share of food supplies and ensures a critical number of ecosystem services all over the world. Though all the countries are rich in producing food, there are also some insecuirities/scarcity of food in the population. This report explains the detail analysis of food production, agricultural land area, food insecurities in people and their undernourishment among the three countries over the period of 8 years (2010-2018).

Indicators:

  - Population in total
  - Food Production index
  - Agricultural Land (% of land area)
  - Prevalence of moderate or severe food insecurity in the population (%)
  - Prevalence of undernourishment (% of population)
  - Permanent Cropland (% of land area)

Time Span: 2010 - 2018
Countries considered: India, Thailand and United States.

In [1]:
import pandas as pd
import plotly
import plotly.express as px
import plotly.graph_objects as go 

In [2]:
DATA_URL = "https://raw.githubusercontent.com/SnigdhaPanakanti/World_Development_Explorer/main/wdi_data.csv"

In [3]:
df = pd.read_csv(DATA_URL)

df.sample(5)

Unnamed: 0  Year       value       indicator Country Code   Country Name  \
35           35  2010   85.630000  AG.PRD.FOOD.XD          IND          India   
162         162  2011    0.295165  AG.LND.CROP.ZS          USA  United States   
104         104  2016   32.903365  AG.LND.ARBL.ZS          THA       Thailand   
71           71  2010   41.222181  AG.LND.AGRI.ZS          THA       Thailand   
42           42  2017  107.770000  AG.PRD.FOOD.XD          IND          India   

                  Region         Income Group    Lending Type  
35            South Asia  Lower middle income            IBRD  
162        North America          High income  Not classified  
104  East Asia & Pacific  Upper middle income            IBRD  
71   East Asia & Pacific  Upper middle income            IBRD  
42            South Asia  Lower middle income            IBRD

**Agricultural Land (% of Land Area):**


- Agricultural Land changes in each country over the span of 2010-2018 is given in the graph below. There is a consistency in the agricultural land area in India throughout the period. United States has a slight decrease in its land percentage during 2011-2013 and then it has increased back to around 40% of the land. Whereas, Thailand have the least area around 10% and increased over the period of 2011-2013 to 40%.


In [4]:
df_agriculture = df.query("Year == 2010").query("indicator == 'AG.LND.AGRI.ZS'")
df_agriculture

Unnamed: 0  Year      value       indicator Country Code   Country Name  \
62          62  2010  60.397418  AG.LND.AGRI.ZS          IND          India   
71          71  2010  41.222181  AG.LND.AGRI.ZS          THA       Thailand   
80          80  2010  44.492513  AG.LND.AGRI.ZS          USA  United States   

                 Region         Income Group    Lending Type  
62           South Asia  Lower middle income            IBRD  
71  East Asia & Pacific  Upper middle income            IBRD  
80        North America          High income  Not classified

In [5]:
df_agri_mean = df_agriculture.groupby('Country Name').mean().reset_index().sort_values(by="value", ascending=False)

df_agri_mean

Country Name  Unnamed: 0    Year      value
0          India        62.0  2010.0  60.397418
2  United States        80.0  2010.0  44.492513
1       Thailand        71.0  2010.0  41.222181

In [ ]:
fig = px.bar(
    df_agri_mean, 
    x="Country Name", 
    labels={"value":"2010 Agricultural land percentage", "Region":""},
    y="value",
    color="Country Name",
    height=800,
    template=list(plotly.io.templates.keys())[5],
    title="Agricultural land (%)"
)

fig.update_layout(showlegend=False)

fig.show()

In [ ]:
fig = px.line(
    df_agri_mean,
    x= "Year",
    labels={"value":"Agricultural land (% of land area)"},
    y= "value",
    height= 700,
    color= "Country Name",
    template=list(plotly.io.templates.keys())[5],
    title= "TimeSeries of Agricultural land (%)"
)

fig.show()

**Permanent Crop Land (% of land area):**

- The permanent cropland in the countries during the decade has no much change except for India where the percentage of permanent cropland land has increased slightly.
- The comparision to Agricultural land and permanent cropland is done to check the percentage of cropland for available agricultural land in each country. The median for the agricultural land is 44.49 and for permanent cropland is 4.11. The observed probability factor (P > |t|) is 0.736.

In [8]:
df_perm_cropland = df.query("indicator == 'AG.LND.CROP.ZS'")

df_perm_cropland.sample(5)

Unnamed: 0  Year     value       indicator Country Code Country Name  \
153         153  2011  8.808158  AG.LND.CROP.ZS          THA     Thailand   
158         158  2016  8.808158  AG.LND.CROP.ZS          THA     Thailand   
144         144  2011  4.167241  AG.LND.CROP.ZS          IND        India   
154         154  2012  8.808158  AG.LND.CROP.ZS          THA     Thailand   
152         152  2010  8.808158  AG.LND.CROP.ZS          THA     Thailand   

                  Region         Income Group Lending Type  
153  East Asia & Pacific  Upper middle income         IBRD  
158  East Asia & Pacific  Upper middle income         IBRD  
144           South Asia  Lower middle income         IBRD  
154  East Asia & Pacific  Upper middle income         IBRD  
152  East Asia & Pacific  Upper middle income         IBRD

In [ ]:
fig = px.line(
    df_perm_cropland,
    x= "Year",
    labels={"value":"Permanent Crop land (% of land area)"},
    y= "value",
    height= 700,
    color= "Country Name",
    template=list(plotly.io.templates.keys())[5],
    title= "Permanent Crop land (% of land area)"
)

fig.show()

**Food Production Index:**

- Food production is the largest source of environmental degradation. To be sustainable, food production must occur within food-related planetary boundaries for climate change, biodiversity loss, land and water use, as well as for nitrogen and phosphorus cycles.

In [10]:
df_food_production = df.query("indicator == 'AG.PRD.FOOD.XD'")

df_food_production.sample(5)

Unnamed: 0  Year   value       indicator Country Code Country Name  \
50          50  2016   95.80  AG.PRD.FOOD.XD          THA     Thailand   
51          51  2017  104.68  AG.PRD.FOOD.XD          THA     Thailand   
39          39  2014   99.49  AG.PRD.FOOD.XD          IND        India   
48          48  2014  105.34  AG.PRD.FOOD.XD          THA     Thailand   
45          45  2011  102.96  AG.PRD.FOOD.XD          THA     Thailand   

                 Region         Income Group Lending Type  
50  East Asia & Pacific  Upper middle income         IBRD  
51  East Asia & Pacific  Upper middle income         IBRD  
39           South Asia  Lower middle income         IBRD  
48  East Asia & Pacific  Upper middle income         IBRD  
45  East Asia & Pacific  Upper middle income         IBRD

-  As a brief, the food production in India, Thailand and USA are shown below.

In [ ]:
fig = px.pie(df_food_production,
             values = 'value',
             names = 'Country Name',
             color = 'Country Name',
             template=list(plotly.io.templates.keys())[5],
            )
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()

- Over the span of 8 years, the countries show a diverse effect in food production each year. Through a line graph in time series, these differences are shown. Though there are multiple ups and downs in the scale of food production over a period, it has an inclination at the end showing the increase in food production in the countries.

In [ ]:
fig = px.line(
    df_food_production,
    x= "Year",
    labels={"value":"Food Production Index (2010)"},
    y= "value",
    height= 700,
    color= "Country Name",
    template=list(plotly.io.templates.keys())[5],
    title= "Food Production Index (2010)"
)

fig.show()

**Food insecurities in the population:**

- Having a diet which is both sufficient in terms of energy (caloric) requirements and diverse to meet additional nutritional needs is essential for good health. Food insecurities among the people is one of the major issues in the world, which inturn leads to the food scarcity. there is a major issue of food insecurities in Thailand compared to India and USA.


In [13]:
df_food_insecurity = df.query("indicator == 'SN.ITK.MSFI.ZS'")

df_food_insecurity

Unnamed: 0  Year  value       indicator Country Code   Country Name  \
27          27  2015   15.1  SN.ITK.MSFI.ZS          THA       Thailand   
28          28  2016   19.7  SN.ITK.MSFI.ZS          THA       Thailand   
29          29  2017   24.8  SN.ITK.MSFI.ZS          THA       Thailand   
30          30  2018   26.4  SN.ITK.MSFI.ZS          THA       Thailand   
31          31  2015   10.5  SN.ITK.MSFI.ZS          USA  United States   
32          32  2016    9.8  SN.ITK.MSFI.ZS          USA  United States   
33          33  2017    8.9  SN.ITK.MSFI.ZS          USA  United States   
34          34  2018    8.4  SN.ITK.MSFI.ZS          USA  United States   

                 Region         Income Group    Lending Type  
27  East Asia & Pacific  Upper middle income            IBRD  
28  East Asia & Pacific  Upper middle income            IBRD  
29  East Asia & Pacific  Upper middle income            IBRD  
30  East Asia & Pacific  Upper middle income            IBRD  
31        North America          High income  Not classified  
32        North America          High income  Not classified  
33        North America          High income  Not classified  
34        North America          High income  Not classified

In [ ]:
fig = px.line(
    df_food_insecurity,
    x= "Year",
    labels={"value":"Food insecurity in the population"},
    y= "value",
    height= 700,
    color= "Country Name",
    template=list(plotly.io.templates.keys())[5],
    title= "Moderate or Severe Food insecurity in the population (%)"
)

fig.show()

**Prevalence of undernourishment:**

- Due to the various reasons like food scarcity, lack of nutritious food, etc., people face undernourishment issues. The prevalence of undernourishment is shown below through a line graph. 

In [15]:
df_undernourishment = df.query("indicator == 'SN.ITK.DEFC.ZS'")

df_undernourishment.sample(5)

Unnamed: 0  Year  value       indicator Country Code   Country Name  \
135         135  2011    2.5  SN.ITK.DEFC.ZS          USA  United States   
120         120  2014   14.9  SN.ITK.DEFC.ZS          IND          India   
140         140  2016    2.5  SN.ITK.DEFC.ZS          USA  United States   
125         125  2010   10.0  SN.ITK.DEFC.ZS          THA       Thailand   
129         129  2014    7.7  SN.ITK.DEFC.ZS          THA       Thailand   

                  Region         Income Group    Lending Type  
135        North America          High income  Not classified  
120           South Asia  Lower middle income            IBRD  
140        North America          High income  Not classified  
125  East Asia & Pacific  Upper middle income            IBRD  
129  East Asia & Pacific  Upper middle income            IBRD

In [ ]:
fig = px.line(
    df_undernourishment,
    x= "Year",
    labels={"value":"Prevalence of undernourishment"},
    y= "value",
    height= 700,
    color= "Country Name",
    template=list(plotly.io.templates.keys())[5],
    title= "Prevalence of undernourishment (%)"
)

fig.show()

**Permanent Cropland - Choropleth:**


In [17]:
df_choropleth = df.query("indicator == 'AG.LND.CROP.ZS'")

df_choropleth.sample(5)

Unnamed: 0  Year     value       indicator Country Code   Country Name  \
150         150  2017  4.372408  AG.LND.CROP.ZS          IND          India   
156         156  2014  8.808158  AG.LND.CROP.ZS          THA       Thailand   
149         149  2016  4.372408  AG.LND.CROP.ZS          IND          India   
165         165  2014  0.295165  AG.LND.CROP.ZS          USA  United States   
167         167  2016  0.295165  AG.LND.CROP.ZS          USA  United States   

                  Region         Income Group    Lending Type  
150           South Asia  Lower middle income            IBRD  
156  East Asia & Pacific  Upper middle income            IBRD  
149           South Asia  Lower middle income            IBRD  
165        North America          High income  Not classified  
167        North America          High income  Not classified

In [ ]:
fig = px.choropleth(
    df_choropleth,
    locations= "Country Code",
    labels={"value":"AG.LND.CROP.ZS"},
    color= "value",
    height= 700,
    color_continuous_scale= "geyser",
    hover_name="Country Name",
    template=list(plotly.io.templates.keys())[5],
    title= "2010 Permanent CropLand (% of land area)"
)

fig.show()

**Correlation between Food Production, Prevalence of Undernourishment and Prevalence of moderate or severe food insecurity:**

In [20]:
df_pivot = df.pivot_table(index = ["Year","Country Code","Country Name","Region","Income Group","Lending Type"],
                         columns = "indicator", values = "value")
df_pivot.head()

indicator                                                                               AG.LND.AGRI.ZS  \
Year Country Code Country Name  Region              Income Group        Lending Type                     
2010 IND          India         South Asia          Lower middle income IBRD                 60.397418   
     THA          Thailand      East Asia & Pacific Upper middle income IBRD                 41.222181   
     USA          United States North America       High income         Not classified       44.492513   
2011 IND          India         South Asia          Lower middle income IBRD                 60.430043   
     THA          Thailand      East Asia & Pacific Upper middle income IBRD                 41.222181   

indicator                                                                               AG.LND.ARBL.ZS  \
Year Country Code Country Name  Region              Income Group        Lending Type                     
2010 IND          India         South Asia          Lower middle income IBRD                 52.808263   
     THA          Thailand      East Asia & Pacific Upper middle income IBRD                 30.848128   
     USA          United States North America       High income         Not classified       17.241638   
2011 IND          India         South Asia          Lower middle income IBRD                 52.798173   
     THA          Thailand      East Asia & Pacific Upper middle income IBRD                 30.848128   

indicator                                                                               AG.LND.CROP.ZS  \
Year Country Code Country Name  Region              Income Group        Lending Type                     
2010 IND          India         South Asia          Lower middle income IBRD                  4.111745   
     THA          Thailand      East Asia & Pacific Upper middle income IBRD                  8.808158   
     USA          United States North America       High income         Not classified        0.295165   
2011 IND          India         South Asia          Lower middle income IBRD                  4.167241   
     THA          Thailand      East Asia & Pacific Upper middle income IBRD                  8.808158   

indicator                                                                               AG.PRD.FOOD.XD  \
Year Country Code Country Name  Region              Income Group        Lending Type                     
2010 IND          India         South Asia          Lower middle income IBRD                     85.63   
     THA          Thailand      East Asia & Pacific Upper middle income IBRD                     96.57   
     USA          United States North America       High income         Not classified           92.42   
2011 IND          India         South Asia          Lower middle income IBRD                     90.87   
     THA          Thailand      East Asia & Pacific Upper middle income IBRD                    102.96   

indicator                                                                               SN.ITK.DEFC.ZS  \
Year Country Code Country Name  Region              Income Group        Lending Type                     
2010 IND          India         South Asia          Lower middle income IBRD                      16.0   
     THA          Thailand      East Asia & Pacific Upper middle income IBRD                      10.0   
     USA          United States North America       High income         Not classified             2.5   
2011 IND          India         South Asia          Lower middle income IBRD                      15.6   
     THA          Thailand      East Asia & Pacific Upper middle income IBRD                       9.2   

indicator                                                                               SN.ITK.MSFI.ZS  \
Year Country Code Country Name  Region              Income Group        Lending Type                     
2010 IND          India         South Asia          Lower middle income I

In [ ]:
indicator = ["AG.PRD.FOOD.XD","SN.ITK.MSFI.ZS",	"SN.ITK.DEFC.ZS"]
fig = px.imshow(df_pivot[indicator].corr(),
                height = 700,
                title = "Correlation Matrix Heatmap (Year 2015)",
                template=list(plotly.io.templates.keys())[5],
                )

fig.show()

**Conclusion:**

- There are various changes in the food production, food nourishment and food insecurity in all the countries. As we have seen the graphs comparing the factors among the countries, there are few changes in the food production over the years. Every country should take care of such concerns like food scarcity/shortage and many more so that people don't die due to hunger.

- It can be determined that the percentage of agricultural land has no bigger changes in India and United States over the period of 8 years. Thailand has an increase in its agricultural land to 40%. The permanent cropland also remained same.